ingestion/src/metadata/ingestion/source/database/snowflake/lineage.py

```diff
@@ -61,3 +66,33 @@ def get_stored_procedure_queries_dict(self) -> Dict[str, List[QueryByProcedure]]
         )
 
         return queries_dict
+
+    def yield_table_query(self) -> Iterator[TableQuery]:
+        """
+        Given an engine, iterate over the query results to
+        yield a TableQuery with query parsing info
+        """
+        for engine in self.get_engine():
+            rows = None
+            with engine.connect() as conn:
+                rows = conn.execute(
+                    self.get_sql_statement(
+                        start_time=self.start,
+                        end_time=self.end,
+                    )
+                ).fetchall()
```

Comment:
this can kill the memory. Try server side cursors. example

```
with engine.connect() as conn:
    result = conn.execution_options(stream_results=True, max_row_buffer=100).execute(
        text("select * from public.employees")
    )
    counter = 0
    for row in result:
        counter += 1
print(counter)
```